extra_data/components.py

```diff
@@ -2013,7 +2013,10 @@ def identify_multimod_detectors(
     res = set()
     for cls in clses:
         for name in cls._find_detector_names(data):
-            res.add((name, cls))
+            if (detector_name is None
+                or detector_name in name
+            ):
+                res.add((name, cls))
```

Comment:
I suggest this change to 'identify_multimod_detectors' function to solve my previous comment.